modules/tinymce/src/plugins/codesample/test/ts/webdriver/CodeSampleCopyAndPasteTest.ts

```diff
@@ -0,0 +1,82 @@
+import { Keys, RealClipboard } from '@ephox/agar';
+import { beforeEach, describe, it } from '@ephox/bedrock-client';
+import { PlatformDetection } from '@ephox/sand';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/codesample/Plugin';
+
+import * as TestUtils from '../module/CodeSampleTestUtils';
+
+describe('webdriver.tinymce.plugins.codesample.CodeSampleCopyAndPasteTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'codesample',
+    toolbar: 'codesample',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ]);
+
+  const pressEnter = (editor: Editor) => {
+    const overrides = { keyCode: Keys.enter() };
+
+    const getInputEventArgs = (eventType: string) => ({ ...new window.InputEvent(eventType), ...overrides });
+    const getKeyboardEventArgs = (eventType: string) => ({ ...new window.KeyboardEvent(eventType), ...overrides });
+
+    editor.dispatch('keydown', getKeyboardEventArgs('keydown'));
+    editor.dispatch('beforeinput', getInputEventArgs('beforeinput'));
+    editor.dispatch('input', getInputEventArgs('input'));
+    editor.dispatch('keyup', getKeyboardEventArgs('keyup'));
+  };
+
+  beforeEach(function () {
+    const browser = PlatformDetection.detect().browser;
+    if (browser.isSafari()) {
+      this.skip();
+    }
```

Comment:
Skipping a browser just because it fails tests isn't the right approach... we need to find out why it's failing and fix it instead of doing this.